Engine/TypeData.cs

```diff
@@ -240,7 +240,10 @@ internal bool IsString
         bool failedLoad;
         TypeData elementType;
         ITypeData baseTypeCache;
-        TypeCode typeCode = TypeCode.Object;
+
+        const int UnloadedTypeCode = 100;
+        
+        TypeCode typeCode = (TypeCode)(UnloadedTypeCode);
```

Comment:
I don't really understand why you added this. When would `type` be non-null if the assembly wasn't loaded?

Reply:
All normal values of TypeCode means something, so we can detect that the typecode has not been set by checking if it is this value.